Client/TabManagement/Tab.swift

```diff
@@ -25,6 +25,7 @@ protocol TabContentScript {
     static func name() -> String
     func scriptMessageHandlerName() -> String?
     func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage)
+    func prepareForDeinit()
```

Comment:
Agree with @razvanlitianu, the empty functions everywhere are a little odd. Can we put an extension with an empty default implementation here so we only need to add it in places that actually use it?